alibi_detect/models/tensorflow/trainer.py

```diff
@@ -57,6 +57,7 @@ def trainer(
     callbacks
         Callbacks used during training.
     """
+    return_xy = False if not isinstance(dataset, tf.keras.utils.Sequence) and y_train is None else True
```

Comment:
This can be probably simplified to `... if not dataset and ...` (would ignore `None` and empty `dataset`) ? Similarly could write ` ... and not y_train ...` to use the same pattern. (That being said I see `if not isinstance(...)` is used on the line below too, so happy to keep this to avoid further changes).